controllers/btpoperator_controller.go

```diff
@@ -520,22 +536,150 @@ func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, s *corev
-			secretIndex = i
+		return false, err
+	}
+
+	logger.Info(fmt.Sprintf("setting namespace of %s secret to %s", clusterIdSecretName, managementNamespace))
+
+	currentConfigMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      btpServiceOperatorConfigMap,
+			Namespace: ChartNamespace,
+		},
+	}
+	if r.Get(ctx, client.ObjectKeyFromObject(&currentConfigMap), &currentConfigMap); err != nil {
+		return false, fmt.Errorf("failed to get config map %s: %w", btpServiceOperatorConfigMap, err)
+	}
+
+	configMapUpdated, err := r.verifyConfigMap(&currentConfigMap, managementNamespace, futureConfigMap)
+	if err != nil {
+		return false, err
+	}
+	if configMapUpdated {
+		operatorSecret.SetNamespace(managementNamespace)
+	}
+
+	secretsValuesMismatch, err := r.verifySecretsIntegrity(managerSecret, &currentConfigMap, logger)
+	if err != nil {
+		logger.Error(err, "while checking cluster id secret integrity")
+		return false, fmt.Errorf("while checking cluster id secret integrity: %w", err)
+	}
+	logger.Info("is clusterId changed", "changed", secretsValuesMismatch)
+
+	return secretsValuesMismatch || configMapUpdated, nil
+}
+
+func (r *BtpOperatorReconciler) verifyConfigMap(currentConfigMap *corev1.ConfigMap, managementNamespace string, futureConfigMap *unstructured.Unstructured) (bool, error) {
+	configMapManagementNamespace, ok := currentConfigMap.Data[managementNamespaceKey]
+	if ok {
+		if managementNamespace != configMapManagementNamespace {
+			if err := unstructured.SetNestedField(futureConfigMap.Object, managementNamespace, "data", "MANAGEMENT_NAMESPACE"); err != nil {
+				return false, err
+			}
+			return true, nil
 		}
 	}
+	if !ok {
+		if err := unstructured.SetNestedField(futureConfigMap.Object, managementNamespace, "data", "MANAGEMENT_NAMESPACE"); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func (r *BtpOperatorReconciler) deleteClusterIDSecret(ctx context.Context, logger *logr.Logger) error {
+	logger.Info("checking cluster id secret integrity")
+	clusterIdSecret := r.buildSecret(clusterIdSecretName, ChartNamespace)
+	err := r.Get(ctx, client.ObjectKeyFromObject(clusterIdSecret), clusterIdSecret)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("cluster id secret not found")
+			return nil
+		}
+		return fmt.Errorf("failed to get secret %s in %s namespace: %w", clusterIdSecret.GetName(), clusterIdSecret.GetNamespace(), err)
+	}
+	err = r.Delete(ctx, clusterIdSecret)
+	if err != nil {
+		return fmt.Errorf("failed to delete secret %s in %s namespace: %w", clusterIdSecret.GetName(), clusterIdSecret.GetNamespace(), err)
+	}
+	return nil
+}
+
+func (r *BtpOperatorReconciler) verifySecretsIntegrity(mangerSecret *corev1.Secret, configMap *corev1.ConfigMap, logger *logr.Logger) (bool, error) {
+	logger.Info("checking cluster id secret integrity")
+	notMatch := !reflect.DeepEqual(mangerSecret.Data[managerSecretClusterIdKey], configMap.Data[operatorSecretClusterIdKey])
```

Comment:
I am not sure if DeepEqual is appropriate. We can have different set of values in secret and config map, and even keys are different (lowercase vs uppercase).